Framework/DataHandling/src/LoadILLPolarizedDiffraction.cpp

```diff
@@ -0,0 +1,411 @@
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidDataHandling/LoadILLPolarizedDiffraction.h"
+#include "MantidAPI/Axis.h"
+#include "MantidAPI/FileProperty.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/RegisterFileLoader.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidAPI/WorkspaceGroup.h"
+#include "MantidGeometry/Instrument/ComponentHelper.h"
+#include "MantidGeometry/Instrument/ComponentInfo.h"
+#include "MantidGeometry/Instrument/DetectorInfo.h"
+#include "MantidKernel/DateAndTime.h"
+#include "MantidKernel/EnabledWhenProperty.h"
+#include "MantidKernel/ListValidator.h"
+#include "MantidKernel/OptionalBool.h"
+#include "MantidKernel/Unit.h"
+#include "MantidKernel/UnitFactory.h"
+#include "MantidKernel/UnitLabelTypes.h"
+#include "MantidKernel/V3D.h"
+#include "MantidKernel/VisibleWhenProperty.h"
+
+#include <Poco/Path.h>
+
+namespace Mantid {
+namespace DataHandling {
+
+using namespace API;
+using namespace Geometry;
+using namespace Kernel;
+using namespace NeXus;
+using Types::Core::DateAndTime;
+
+namespace {
+// This defines the number of detector banks in D7
+constexpr size_t D7_NUMBER_BANKS = 3;
+// This defines the number of physical pixels in D7
+constexpr size_t D7_NUMBER_PIXELS = 132;
+// This defines the number of pixels per bank in D7
+constexpr size_t D7_NUMBER_PIXELS_BANK = 44;
+// This defines the number of monitors in the instrument. If there are cases
+// where this is no longer one this decleration should be moved.
+constexpr size_t NUMBER_MONITORS = 2;
+// The conversion factor from radian to degree
+constexpr double DEG_TO_RAD = M_PI / 180.;
+} // namespace
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_NEXUS_FILELOADER_ALGORITHM(LoadILLPolarizedDiffraction)
+
+/// Returns confidence. @see IFileLoader::confidence
+int LoadILLPolarizedDiffraction::confidence(NexusDescriptor &descriptor) const {
+
+  // fields existent only at the ILL Diffraction
+  if (descriptor.pathExists("/entry0/D7")) {
+    return 80;
+  } else {
+    return 0;
+  }
+}
+
+/// Algorithms name for identification. @see Algorithm::name
+const std::string LoadILLPolarizedDiffraction::name() const {
+  return "LoadILLPolarizedDiffraction";
+}
+
+/// Algorithm's version for identification. @see Algorithm::version
+int LoadILLPolarizedDiffraction::version() const { return 1; }
+
+/// Algorithm's category for identification. @see Algorithm::category
+const std::string LoadILLPolarizedDiffraction::category() const {
+  return "DataHandling\\Nexus;ILL\\Diffraction";
+}
+
+/// Algorithm's summary for use in the GUI and help. @see Algorithm::summary
+const std::string LoadILLPolarizedDiffraction::summary() const {
+  return "Loads ILL polarized diffraction nexus files.";
```

Comment:
This algorthms code is specific to only load D7 files as far as I can see, the summary should state this.